test/CMakeLists.txt

```diff
@@ -74,23 +74,31 @@ else() # Download CRTM coefficients
   set(test_files_dirname ${CRTM_COEFFS_BRANCH}.tgz) 
   set(checksum "58e0a5c698a438a31dc4914fcda39846") # MD5SUM of fix_REL-3.1.1.2.tgz
   
-  set(DOWNLOAD_URL "${DOWNLOAD_BASE_URL}/${test_files_dirname}")
-  set(DOWNLOAD_DEST "${CRTM_COEFFS_PATH}/${test_files_dirname}")
-  
-  # Download during CMake configure step
-  message(STATUS "Downloading CRTM coeffs files from: ${DOWNLOAD_URL} to ${DOWNLOAD_DEST}")
-  file(DOWNLOAD ${DOWNLOAD_URL} ${DOWNLOAD_DEST}
-    EXPECTED_MD5 ${checksum}
-    SHOW_PROGRESS)
+  # Check if the CRTM binary tarball is already present otherwise download it.
+  if (DEFINED ENV{CRTM_BINARY_FILES_TARBALL})
+    file(MD5 "$ENV{CRTM_BINARY_FILES_TARBALL}" computed_checksum )
+    if (NOT "${computed_checksum}" STREQUAL "${checksum}")
+      message(FATAL_ERROR "Environment provided CRTM_BINARY_FILES_TARBALL=$ENV{CRTM_BINARY_FILES_TARBALL} had checksum '${computed_checksum}' not matching expected '${checksum}'")
+    endif()
```

Comment:
I like the idea of crashing if the user provides the wrong tarball. However, it's important to check if others want that behavior too or they may want to be nice trying to download the new one.

Reply:
As a general rule, it's better to crash with a clear error rather than quietly do something other than what users think they are doing.